### PR TITLE
arch/arm/samv7: enforce compilation error for incompatible configuration options

### PR DESCRIPTION
**Pull request by pkarashchenko**
## Summary
`CONFIG_SAMV7_JTAG_FULL_ENABLE` and `CONFIG_SAMV7_USART1` can't be enabled at the same time since both use the same pin

## Impact
Improve SAMv7 experience

## Testing
Pass CI

### CODE REVIEW

File: arch/arm/src/samv7/sam_serial_spi.c [hunk 1]
```diff
@@ -69,16 +69,16 @@
 
 struct sam_serial_spi_pins_s
 {
-  uint32_t mosi;
-  uint32_t miso;
-  uint32_t sck;
-  uint32_t nss;
+  const uint32_t mosi;
```
Comment (xiaoxiang781216) [inline arch/arm/src/samv7/sam_serial_spi.c:72]: why add const to integer fields?

File: arch/arm/src/samv7/sam_serial_spi.c [hunk 2]
```diff
@@ -69,16 +69,16 @@
...
+  const uint32_t miso;
+  const uint32_t sck;
+  const uint32_t nss;
 };
 
 struct sam_serial_spi_s
 {
   struct sam_serial_spi_pins_s pins;
-  uint32_t base;               /* SPI controller register base address */
+  const uint32_t base;         /* SPI controller register base address */
```
Comment (xiaoxiang781216) [inline arch/arm/src/samv7/sam_serial_spi.c:81]: ditto

**Comment by michallenc** [comment]
Should not this be already set in https://github.com/apache/nuttx/blob/master/arch/arm/src/samv7/sam_gpio.c#L466? And just raise compile time error if both `USART1` and `CONFIG_SAMV7_JTAG_FULL_ENABLE` is set?

File: arch/arm/src/samv7/sam_serial_spi.c (hunk 1 above)
Comment (pkarashchenko) [reply arch/arm/src/samv7/sam_serial_spi.c:72]: I can revert. It was just from logical perspective as the values of the pin should not change

File: arch/arm/src/samv7/sam_serial_spi.c (hunk 2 above)
Comment (pkarashchenko) [reply arch/arm/src/samv7/sam_serial_spi.c:81]: I can revert. It was just from logical perspective as the values of the base address should not change

**Comment by pkarashchenko** [comment]
> Should not this be already set in https://github.com/apache/nuttx/blob/master/arch/arm/src/samv7/sam_gpio.c#L466? And just raise compile time error if both `USART1` and `CONFIG_SAMV7_JTAG_FULL_ENABLE` is set?

I didn't notice the code in `sam_gpio.c` and only found https://github.com/apache/nuttx/blob/4b2b26c0b05f4c034d781a6215971e1675623486/arch/arm/src/samv7/sam_lowputc.c#L335
Indeed, maybe it is a good idea to rise an error. I've checked my config and indeed the `CONFIG_SAMV7_JTAG_FULL_ENABLE` is set in my case, so rising an error may fix the inconsistency in the code

File: arch/arm/src/samv7/sam_lowputc.c [hunk 3]
```diff
@@ -327,14 +327,10 @@ void sam_lowsetup(void)
    */
 
 #  if defined(CONFIG_SAMV7_JTAG_FULL_ENABLE)
-#    warning CONFIG_SAMV7_JTAG_FULL_ENABLE is incompatible with CONFIG_SAMV7_USART1.
-#    warning The SYSIO Pin4 must be bound to PB4 to use USART1
+#    error CONFIG_SAMV7_JTAG_FULL_ENABLE is incompatible with CONFIG_SAMV7_USART1.
```
Comment (michallenc) [inline arch/arm/src/samv7/sam_lowputc.c:330]: This should now be redundant with the same check in `sam_gpio.c`, no?

Comment (pkarashchenko) [reply arch/arm/src/samv7/sam_lowputc.c:330]: Ok. I can remove one from here

**Review by michallenc — APPROVED** [review]
LGTM

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**